cirq-core/cirq/contrib/acquaintance/mutation_utils.py

```diff
@@ -98,14 +98,19 @@ def replace_acquaintance_with_swap_network(
     return reflected
 
 
-class ExposeAcquaintanceGates(optimizers.ExpandComposite):
+class ExposeAcquaintanceGates:
     """Decomposes permutation gates that provide acquaintance opportunities."""
 
     def __init__(self):
-        circuits.PointOptimizer.__init__(self)
         self.no_decomp = lambda op: (
             not get_acquaintance_size(op) or isinstance(op.gate, AcquaintanceOpportunityGate)
         )
 
+    def optimize_circuit(self, circuit: 'cirq.Circuit') -> None:
```

Comment:
Does this fulfill the `Callable[[CIRCUIT_TYPE, TransformerContext], CIRCUIT_TYPE]` type if it doesn't specify the context argument?

Similarly for other transformers inheriting from `expand_composite`.

Reply:
- The newly added `expand_composite` transformer is a standalone method now, not a class, and it satisfies the new transformer API interface. 
- `ExposeAcquaintanceGates`, and other modified optimizers in contrib/*, have been changed to call `expand_composite` under the hood in their implementation instead of deriving from the original `ExpandComposite` class, which is now deprecated. 
- Note that these optimizers in contrib, like`ExposeAcquaintanceGates`, are not valid cirq transformers and would not satisfy the transformer API interface expected by `@cirq.transformer` decorator. This is the reason they are not decorated with `@cirq.transformer`.

Reply:
Is there a motivation for **not** converting contrib optimizers to transformers? From my perspective, adding an unused `context` arg to their `__call__` method is cheap and non-breaking.